### ISSUE TITLE
Is this legal?

### ISSUE THREAD

**Issue by astralbat** [r-maerz/archlinux-lts-zfs#5]
First: much appreciated, I have made use of the release to sidestep some of my problems getting zfs to work on a recent official arch ISO. The instructions would probably have been helpful for me on their own.

But isn't this providing what arch cannot do because of the licensing conflict between zfs and the kernel. Now they're being distributed togther?

**Comment by irreleph4nt** [comment]
Glad you found this project useful.
TBH, I am not a lawyer, so I depent on people smarter than me for such questions. What I have found is that ... 

- the legal problems arise mostly from GPLv2 (linux kernel) requiring components included in GPLv2 projects to be licensed as GPLv2 themselves, whilst CDDL of ZFS is more lenient
- the OpenZFS project itself advises shipping the ZFS binary / module side by side with the kernel is fine, as neither source is directly included in the other binary (see [1])
- Canonical has been shipping ZFS with Ubuntu for a number of years now and has not been sued for it (see [2])

So all in all, I will probably update my readme a bit to call out this is first and foremost a project I started for myself. But then again, given the public nature of GitHub repos, I have no reasonable way of preventing others from making use of it, too.

[1] https://openzfs.github.io/openzfs-docs/License.html
[2] https://www.reddit.com/r/zfs/comments/1cjj2mh/can_somebody_eli5_why_other_distro_dont_include/

**Comment by r-maerz** [comment]
Glad you found this project useful.
TBH, I am not a lawyer, so I depent on people smarter than me for such questions. What I have found is that ...

- the legal problems arise mostly from GPLv2 (linux kernel) requiring components included in GPLv2 projects to be licensed as GPLv2 themselves, whilst CDDL of ZFS is more lenient

- the OpenZFS project itself advises shipping the ZFS binary / module side by side with the kernel is fine, as neither source is directly included in the other binary (see [1])

- Canonical has been shipping ZFS with Ubuntu for a number of years now and has not been sued for it (see [2])

So all in all, I will probably update my readme a bit to call out this is first and foremost a project I started for myself. But then again, given the public nature of GitHub repos, I have no reasonable way of preventing others from making use of it, too.

[1] https://openzfs.github.io/openzfs-docs/License.html
[2] https://www.reddit.com/r/zfs/comments/1cjj2mh/can_somebody_eli5_why_other_distro_dont_include/
